CodeEdit/Features/NavigatorSidebar/SourceControlNavigator/SourceControlToolbarBottom.swift

```diff
@@ -7,9 +7,42 @@
 
 import SwiftUI
 
+// Runs the specified shell command
+@discardableResult
+func shell(_ command: String) -> String {
+    let task = Process()
+    let pipe = Pipe()
+
+    task.standardOutput = pipe
+    task.standardError = pipe
+    task.arguments = ["-c", command]
+    task.launchPath = "/bin/zsh"
+    task.standardInput = nil
+    task.launch()
+
+    let data = pipe.fileHandleForReading.readDataToEndOfFile()
+    let output = String(data: data, encoding: .utf8)!
+
+    return output
+}
+
+// Gets the current workspace's path
+func getCurrentWorkspaceDocument(workspace: WorkspaceDocument) -> String {
+    return String(String(describing: workspace.fileURL!.absoluteURL).dropFirst(7))
+}
+
+// Changes to true when the user has an commit the needs to be pushed
+var commited: Bool = false
```

Comment:
1. We have a shell client available already - no need to declare another one.
2. I'm pretty sure that the workspace already has a property which exposes the base path of the workspace.
3. Don't implement unscoped global vars or funcs

Reply:
@lukepistrol Thanks for the review, what's the shell client that available already? Are you referring to the SwiftTerm terminal, if so I can't send a command there. I'll fix every other issue that you've highlighted.

Reply:
CodeEdit/Utils/ShellClient/Models/ShellClient.swift

Reply:
> CodeEdit/Utils/ShellClient/Models/ShellClient.swift

Thanks for that, BTW as far as I can see there isn't a better way to find the path and to remove the (unwanted) `file://` at the start.

Reply:
https://developer.apple.com/documentation/foundation/url#2902598

Reply:
I've looked (everywhere?) on the linked documentation and I can't find anything that could help me.

Reply:
1 and 3 are fixed.